### ISSUE TITLE
d3-selection-multi Definitions

### ISSUE THREAD

**Issue by ocombe** [tomwanzek/d3-v4-definitelytyped#33]
A lot of methods from Selection support an object as a param. Example:

``` ts
d3.selectAll('div')
  .attr({
    position: 'absolute',
    top: 0
  });
```

The definitions of these functions (attr, style, classed, ...) should allow this object parameter (type any ?).

**Comment by tomwanzek** [comment]
That is a change from d3 v3 to d3-selection. Outside the D3 v4 modules which make up the default bundle there is a separate module d3-selection-multi. I believe that is where the additional functionality resides. I have not looked at it yet (as it is not in the default bundle set).

I tend to recall that there are related issues on D3 which Mike has answered.

**Comment by ocombe** [comment]
Yeah so these functions have been renamed with an "s": `.attr({ ... })` is now `.attrs`, same thing for the others (styles, ...).
`.attr` doesn't support object parameters anymore !

**Comment by tomwanzek** [comment]
Just for reference, I am going to reopen this issue. Rename it to 'd3-selection-multi definitions' (Not default bundle).

I am placing it into _parking lot_ for the time being to focus on the other main items.

**Comment by gustavderdrache** [comment]
Are these definitions still up for grabs? I'd love to roll up my sleeves and pitch in, if you'd allow me.

**Comment by tomwanzek** [comment]
@gustavderdrache Totally welcome and most appreciated. @Ledragon is working away on d3-geo, I have a couple of follow-ups to do there as well. So, yes, if you want to tackle d3-selection-multi that would be great. I think the augmentation pattern would be similar to what d3-transition does for d3-selection. With the exception that d3-selection-multi does not have actual exports of its own.

There is a bit of a contribution guide for this repo.

I think @Ledragon also had mentioned a starting point/interest in contributing d3-scale-chromatic.

What I tend to do is this: After a module is complete with tests in this repo and included in a pre-release, I move it through my DefinitelyTyped fork as a PR against types-2.0.

We have completed versions of d3-dsv and d3-request in this repo as well. I have not PRed them to DT yet, as we needed to address the legacy issue (d3-dsv). I saw your comment in regard to versioning over on the migration issue.  I will add on  to that. I think my main issue is the missing feedback from the TypeScript folks on the migration path between the `master` and `types-2.0` branches on DT (I.e. the synching) and the UMD module pattern for the `d3` global. 

So in the meantime we just focused on getting through the  modules until we can complete the migration. I thought about adding a couple more of the "optional modules" on, once the main work is done. I scanned through the D3 org and picked of some that are led by Mike Bostock. I am going to tag the suitable ones to the optional scope section when I have a second.
